README.md

```diff
@@ -2,4 +2,107 @@
 
 [![License badge](https://img.shields.io/badge/license-GPL-blue.svg)](https://opensource.org/licenses/GPL-3.0)
 [![Docker badge](https://img.shields.io/docker/pulls/dojot/iotagent-json.svg)](https://hub.docker.com/r/dojot/iotagent-json/)
-[![Support badge]( https://img.shields.io/badge/support-sof-yellowgreen.svg)](http://stackoverflow.com/questions/tagged/dojot)
+[![Travis badge](https://travis-ci.org/dojot/iotagent-json.svg?branch=cpqd_master)](https://travis-ci.org/dojot/iotagent-json#)
+
+IoT agents are responsible for receiving messages from physical devices (directly or through a gateway) and sending them commands in order to configure them. This iotagent-json, in particular, receives messages via MQTT with JSON payloads.
+
+## How does it work
+The iotagent-json depends on two things: a Kafka broker, so that it can receive messages informing it about new devices (and, in extension, about their updates and removals), and, obviously, a MQTT broker, so that it can receive messages from the devices. It waits for messages sent through these two elements: from device manager with a management operation on a device and from the MQTT broker with a message sent by a device.
+
+### MQTT
+MQTT is a somewhat simple protocol: it follows a publish/subscriber paradigm and messages are sent to and fro through topics. These topics are simple strings containing tokens separated by forward slashes, such as "/admin/cafe/attrs". A publisher can, well, publish messages by sending them to a MQTT broker using a particular topic and all the subscribers that are listening to that topic will receive a copy of the message.
```

Comment:
> are sent to and fro through topics.

from or through?

> These topics are simple strings containing tokens separated by forward slashes

the topics are simple strings, their labels/names/addresses are